### PR TITLE
Use conan-io/hello example for the new command

### LINKED ISSUE #4916: Move memsharded/hello repository to the conan-io organization
When using the ``conan new`` command, the conanfile.py created points to https://github.com/memsharded/hello and it is a bit weird to use an example out of the organization.

We should move it to a isolated repo in the conan-io org and give it a better name, like: hello-cpp-lib-example

After that, the ``conan new`` command should be updated to use the new repo

### PR DESCRIPTION
**Pull request by danimtb**
Changelog: Feature: Updated the generated *conanfile.py* in ``conan new`` to the new [conan-io/hello].(https://github.com/conan-io/hello) repository
Docs: https://github.com/conan-io/docs/pull/1269

- [x] Refer to the issue that supports this Pull Request: closes #4916 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Comment by jgsogo** [comment]
There could be some places in the docs where we are using this `conan new ....` and then having a look at the generated `conanfile.py`... it could be confusing if the user calls the command and have a different generated conanfile (probably it's not so important)...

**Comment by danimtb** [comment]
I have included the changes in the documentation too

File: conans/client/cmd/new.py [hunk 1]
```diff
@@ -21,13 +21,13 @@ class {package_name}Conan(ConanFile):
     generators = "cmake"
 
     def source(self):
-        self.run("git clone https://github.com/memsharded/hello.git")
-        self.run("cd hello && git checkout static_shared")
+        self.run("git clone https://github.com/conan-io/hello.git")
+        self.run("cd hello")
```
Comment (memsharded) [inline conans/client/cmd/new.py]: useless ``cd hello``

File: conans/test/integration/devflow_test.py [hunk 2]
```diff
@@ -244,7 +244,7 @@ def build_local_different_folders_test(self):
         install_dir = os.path.join(client.current_folder, "install_x86_64")
         tools.replace_in_file(os.path.join(client.current_folder, "src", "hello", "CMakeLists.txt"),
                               "${CMAKE_BINARY_DIR}/conanbuildinfo.cmake",
-                              '"%s/conanbuildinfo.cmake"' % install_dir,
```
Comment (memsharded) [inline conans/test/integration/devflow_test.py]: why necessary? It is not failing so far

Comment (danimtb) [reply conans/test/integration/devflow_test.py]: I was wondering the same and I have no idea...

Comment (jgsogo) [reply conans/test/integration/devflow_test.py]: It failed in [this CI job](https://conan-ci.jfrog.info/blue/organizations/jenkins/ConanTestSuite/detail/PR-5069/2/pipeline), I would want to know what's happening, but if it is needed, this PR is ok for me and ready to merge.

**Pull request merged**